src/synthesizer/parametric/sf_hist.py

```diff
@@ -536,3 +571,94 @@ def _sfr(self, age):
         term1 = (age / self.tau) ** self.alpha
         term2 = (age / self.tau) ** self.beta
         return (term1 + term2) ** -1
+
+
+class DenseBasis(Common):
+    """
+    Dense Basis representation of a SFZH.
+
+    See here for more details on the Dense Basis method.
+
+    https://dense-basis.readthedocs.io/en/latest/
+
+    Attributes:
+        db_tuple (tuple)
+            Dense basis parameters describing the SFH
+            1) total mass formed
+            2) SFR at the time of observation (see redshift)
+            3) number of tx parameters
+            4) times when the galaxy formed fractions of its mass,
+                units of fraction of the age of the Universe [0-1]
+        redshift (float)
+            redshift at which to scale the SFH
+    """
+
+    def __init__(self, db_tuple, redshift):
+
+        # Initialise the parent
+        Common.__init__(
+            self, name="DenseBasis", db_tuple=db_tuple, redshift=redshift
+        )
+
+        self.db_tuple = db_tuple
+        self.redshift = redshift
+
+        # convert dense basis parameters (1db_tuple`) into SFH
+        self._convert_db_to_sfh()
+
+    def _convert_db_to_sfh(
+        self, interpolator="gp_george", min_age=5, max_age=10.3
+    ):
+        """
+        Convert dense basis representation to a binned SFH
+
+        Args:
+            interpolator (string)
+                Dense basis interpolator to use. Options: [gp_george,
+                gp_sklearn, linear, and pchip]
```

Comment:
Would dropping gp_sklearn as an option allow us to remove `scikit-learn` as a dependency?

Reply:
Looking at the failed action I'm guessing the answer to this is a resounding no 😂 

Shall we just live with the extra dependencies for now? I forget what it is but there was something a while back that could have benefited from `scikit-learn` but I avoided it. We may just have to have a serious think about dependencies soon, the list is pretty long now.

Reply:
as discussed offline, waiting on Kartheik to review a change to `dense_basis` to make the sklearn dependency optional

Reply:
Fantastic!